### PR TITLE
Add support for running the autopilot on a different machine than gazebo

### PR DESCRIPTION
**Pull request by tfoote**
Previously the assumption was made that gazebo and the fdm shared an address. 

This is currently based upon my proposed extension, (https://github.com/khancyr/ardupilot/pull/1,  to #4883

But can also be rebased without it relatively easily. Only the last commit is different: ed0408a6e3cd1fa5b742f69f89ae46fe6ac0a5c6

### CODE REVIEW

**Comment by rmackay9** [comment]
@tfoote, thanks for this.  I suspect that @khancyr, @OXINARF or @peterbarker is the best one to give the go/no-go on this one.

On a general note, I'm keen to improve our ROS support so if you hear of people having issues with ardupilot integration with ROS, feel free to send them my way.  They can do that by either creating an issue in our issues list (https://github.com/ArduPilot/ardupilot/issues) and put ROS in the title or they can chat to me directly in our VisionProjects gitter room (https://gitter.im/ArduPilot/ardupilot/VisionProjects).

This is an example of an issue that was raised this morning by a ROS user that I'm attempting to address: https://github.com/ArduPilot/ardupilot/pull/5345

**Comment by tfoote** [comment]
Thanks @rmackay9 I started one new issue here at #5347 

I'll note that this allows you to scale simulating multiple drones in one environment beyond one computer. We're shooting to simulate 100 drones simultaneously. With this we can run N autopilots over M computers connecting to one central gazebo instance.

File: libraries/SITL/SIM_Aircraft.cpp [hunk 1]
```diff
@@ -406,23 +406,21 @@ void Aircraft::set_speedup(float speedup)
 /*
   Create and set in/out socket
 */
-void Aircraft::set_interface_ports(const char* address, const int port_in, const int port_out)
+void Aircraft::set_interface_ports(const char* local_address, const char* gazebo_address, const int port_in, const int port_out)
```
Comment (khancyr) [inline libraries/SITL/SIM_Aircraft.cpp]: this should be const char* gazebo_address but simulator_address to be generic, as ideal the set_interface_ports should be use by every SITL plugin

File: libraries/SITL/SIM_Aircraft.cpp [hunk 2]
```diff
@@ -406,23 +406,21 @@ void Aircraft::set_speedup(float speedup)
...
 {
-    if (!socket_in.bind("127.0.0.1", port_in)) {
-        fprintf(stderr, "SITL: socket in bind failed on sim in : %d  - %s\n", port_in, strerror(errno));
-        fprintf(stderr, "Abording launch...\n");
```
Comment (khancyr) [inline libraries/SITL/SIM_Aircraft.cpp]: I would like to keep fprintf(stderr, "Abording launch...\n"); that way it is sure the launch has quit!

**Comment by tfoote** [comment]
Good points. Updated in ed0408a6e3cd1fa5b742f69f89ae46fe6ac0a5c6

**Comment by khancyr** [comment]
@tfoote Hello ! Do you to rebase you work or you want me to do it in another PR ?

**Comment by khancyr** [comment]
Closed as merged as part of https://github.com/ArduPilot/ardupilot/pull/4883

**Pull request closed without merging**